Common server role

jbzdak opened the pull request:
A base role for all our deployments (hopefully!), by itself it installs etckeeper and tmpreaper. But mostly it is here to enforce installation of dependant roles: 
-  configure-apt
- geerlingguy.security
- kamaln7.swapfile
- sanity-checker - almost every server has something to monitor, even if this is a disk space on root partition.

antoviaque commented on defaults/main.yml:
```diff
@@ -0,0 +1,3 @@
+
+ETCKEEPER_COMMIT_USER: "Etc Keeper"
+ETCKEEPER_COMMIT_EMAIL: "xavier@opencraft.com"
```
ops@opencraft.com

Also, worth reusing a more global variable here, since this would be needed in multiple places?

We might also want to set the default to null@example.com, so we don't end up receiving the notifications of people reusing the roles, if any : )

antoviaque commented on tasks/etckeeper.conf:
```diff
@@ -0,0 +1,19 @@
+export GIT_AUTHOR_NAME="etckeeper"
```
This file is in the wrong folder - use `files/`

antoviaque commented on tasks/etckeeper.conf:
```diff
@@ -0,0 +1,19 @@
+export GIT_AUTHOR_NAME="etckeeper"
+export GIT_AUTHOR_EMAIL="root@localhost"
+export GIT_COMMITTER_NAME=$GIT_AUTHOR_NAME
+export GIT_COMMITTER_EMAIL=$GIT_AUTHOR_EMAIL
```
Why not reuse the same value as for etckeeper, as a template?

antoviaque commented:
:+1:

Commit:
Common server role

Squashed commit of the following:

commit 9e7bbd617e659532f22f27b265d8ed99be15d512
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Fri Apr 29 21:13:57 2016 +0200

    Remove forward-server-mail from common server role

commit 8dba37cc96d2f90225349c35e5e4ecd9b80b3db3
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Tue Apr 26 17:55:35 2016 +0200

    Added short hostname to hosts

commit ed16557fa0cdd3edcc302a6731d1e3df13a65a37
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Tue Apr 26 17:50:30 2016 +0200

    Fixed issue with recursive variable

commit 30214f3f16606da2a370ee3a91641aa18a0cabf8
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Tue Apr 26 17:44:07 2016 +0200

    Make default common server role work with nonmodified ubuntu images.

commit c5c7cdcdf470300d7bf7396b6cfbc72354b70211
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Mon Apr 25 20:53:22 2016 +0200

    Added switch to disable backups

commit 5c5f2059fd9138fea0c77741aa093b6e1d5cab4d
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Fri Apr 22 00:15:11 2016 +0200

    Requirements fix

commit 412be9cb8addaf7abded0d95ed7b12fdda96e74c
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Thu Apr 21 23:09:36 2016 +0200

    Removed updating hosts

commit 48bd9556ebc9f81f3d736914c731ad51ddd3facd
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Thu Apr 21 22:50:34 2016 +0200

    Small fix

commit 537ef5aa6c46f9e6ef88c457990dc43a77017622
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Thu Apr 21 22:39:35 2016 +0200

    Temp debug

commit 236fdc36198bd9a2ef281b2f269945106915a00f
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Thu Apr 21 22:29:10 2016 +0200

    Added seting hostname

commit 7d6b75af817bd5cae88f18322bcc94c0cbd11de2
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Thu Apr 21 20:25:17 2016 +0200

    Changed email address

commit acd24883606a43943d11d6cb747a656e8d48384c
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Thu Apr 21 20:22:06 2016 +0200

    Renamed tarsnap role

commit c810b14501822e2f6d3f16a3fe9bd77c23b42217
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Thu Apr 21 19:30:59 2016 +0200

    Added ability to set hosts files

commit 7e996f9d37d81a254411c6b2a6bbff6fb05abe35
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Thu Apr 21 15:50:00 2016 +0200

    Conditionally enavle tarsnap backup

commit a97f4288175590938bd209a3b160af0d8be7c067
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Wed Apr 20 21:42:08 2016 +0200

    Added tarsnap role

commit 21475c7dee4803fecb6301854504937134fc847b
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Wed Apr 20 09:02:16 2016 +0200

    Fix etckeeper config

commit 2a05a74cf33a4ba618b3f5914d378a7e31ac61d9
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Tue Apr 19 23:28:23 2016 +0200

    Small fix

commit 12501cab02d89175292de52208a74040ab4f30a3
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Tue Apr 19 22:45:36 2016 +0200

    PR review fixes

commit 7a85e8d9f193a99abf6bf545c8ce739b01fd7cd6
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Tue Apr 19 22:40:28 2016 +0200

    PR fixes

commit 4e4929ea2719dafca4d4ee1f55738967f2cdc5c6
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Mon Apr 18 17:33:39 2016 +0200

    Removed unneded dependency

commit b5c68ab678627825dc61ae4e5bd43c625c792a8e
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Mon Apr 18 14:09:25 2016 +0200

    Renamed dependency

commit 726d0c556424903b70f2cbdc1e033692dd989658
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Mon Apr 18 14:04:51 2016 +0200

    Renamed a dependant role

commit cacbb6d0b15a874626aa5dc7b8a132e39e38ee0d
Author: Jacek Bzdak <jacek@opencraft.com>
Date:   Mon Apr 18 13:09:22 2016 +0200

    Common server role

jbzdak commented:
Squashed and merging.
